### PR TITLE
Add utility dataclasses for working with (N)EVR

### PR DESCRIPTION
**Pull request by nforro**
This will also be useful in packit for working with NEVRs of Koji builds.

### CODE REVIEW

File: specfile/utils.py [hunk 1]
```diff
@@ -1,10 +1,55 @@
...
+    def __str__(self) -> str:
+        epoch = f"{self.epoch}:" if self.epoch > 0 else ""
+        release = f"-{self.release}" if self.release else ""
+        return f"{epoch}{self.version}{release}"
+
+    @staticmethod
+    def from_string(evr: str) -> "EVR":
+        m = re.match(r"^(?:(\d+):)?(.*?)(?:-([^-]*))?$", evr)
+        if not m:
+            return EVR()
```
Comment (FrNecas) [inline specfile/utils.py:25]: Is an "empty version" semantically valid? I'd maybe prefer throwing an error here.

Comment (nforro) [reply specfile/utils.py:25]: Hm, I thought it was, but it isn't. Let me fix that.

File: specfile/utils.py [hunk 2]
```diff
@@ -1,9 +1,54 @@
...
 import urllib.parse
+from dataclasses import dataclass
 from pathlib import Path
 
+from specfile.exceptions import SpecfileException
+
+
+@dataclass(kw_only=True)
+class EVR:
+    epoch: int = 0
```
Comment (TomasTomecek) [inline specfile/utils.py:14]: please document what EVR stands for so it's clear for everyone who's not familiar with this shortcut

File: specfile/utils.py [hunk 3]
```diff
@@ -1,9 +1,54 @@
...
+    version: str
+    release: str = ""
+
+    def __str__(self) -> str:
+        epoch = f"{self.epoch}:" if self.epoch > 0 else ""
+        release = f"-{self.release}" if self.release else ""
+        return f"{epoch}{self.version}{release}"
+
+    @staticmethod
+    def from_string(evr: str) -> "EVR":
```
Comment (TomasTomecek) [inline specfile/utils.py:24]: what's the benefit here using staticmethod over classmethod?

File: specfile/utils.py [hunk 4]
```diff
@@ -1,9 +1,54 @@
...
+        m = re.match(r"^(?:(\d+):)?(.*?)(?:-([^-]*))?$", evr)
+        if not m:
+            raise SpecfileException("Invalid EVR string.")
+        e, v, r = m.groups()
+        return EVR(epoch=int(e) if e else 0, version=v, release=r or "")
+
+
+@dataclass(kw_only=True)
+class NEVR:
+    name: str
```
Comment (TomasTomecek) [inline specfile/utils.py:34]: dtto :)

**Review by TomasTomecek — COMMENTED** [review]
very nice utility class

**Review by FrNecas — APPROVED** [review]

**Review by TomasTomecek — APPROVED** [review]

**Pull request merged**